File: yh_tutorial/yh_tutorial_2/src/yh_pub_2.cpp
#include "ros/ros.h"
#include "yh_tutorial_2/yh_msg_2.h"

int main(int argc, char **argv)
{
    ros::init(argc, argv, "yh_pub_2");
    ros::NodeHandle nh;

    ros::Rate loop_rate(5);
    int cnt = 0;

    ros::Publisher pub = nh.advertise<yh_tutorial_2::yh_msg_2>("yh_topic_2", 100);
    yh_tutorial_2::yh_msg_2 msg;

    while (ros::ok())
    {
        msg.stamp = ros::Time::now();
        msg.data = cnt;
        pub.publish(msg);
        cnt++;
        loop_rate.sleep();
    }

    return 0;
}

File: yh_tutorial/yh_tutorial_2/scripts/yh_pub_2.py
#!/usr/bin/python
# -*- coding: utf8 -*-

import rospy
from yh_tutorial_2.msg import yh_msg_2

def talker():
    pub = rospy.Publisher("yh_topic_2", yh_msg_2, queue_size=100)
    rospy.init_node("yh_pub_2", anonymous=True)

    loop_rate = rospy.Rate(5)
    msg = yh_msg_2()
    cnt = 0

    while not rospy.is_shutdown():
        msg.stamp = rospy.get_rostime()
        msg.data = cnt
        pub.publish(msg)
        cnt += 1
        loop_rate.sleep()
    
if __name__ == "__main__":
    try:
        talker()
    except rospy.ROSInterruptException:
        pass